atecc608a_se.c

```diff
@@ -161,6 +163,43 @@ static psa_status_t atecc608a_export_public_key(psa_key_slot_number_t key,
     atecc608a_deinit();
     return status;
 }
+static psa_status_t atecc608a_import_public_key(psa_key_slot_number_t key_slot,
+                                                psa_key_lifetime_t lifetime,
+                                                psa_key_type_t type,
+                                                psa_algorithm_t alg,
+                                                psa_key_usage_t usage,
+                                                const uint8_t *p_data,
+                                                size_t data_length)
+{
+    const uint16_t key_id = key_slot;
+    psa_status_t status = PSA_ERROR_GENERIC_ERROR;
+
+    /* Keys 8 to 15 can store public keys. Slots 1-7 are too small.
+       We also check if the key has a size of 65 {0x04, X, Y}. */
+    if (key_id < 7 || key_id > 15 || data_length != ATCA_PUB_KEY_SIZE + 1)
```

Comment:
What we care to check is that the input to this function is in PSA format for the only algorithm we support. We should use `PSA_ASYMMETRIC_SIGN_OUTPUT_SIZE()` instead of `ATCA_PUB_KEY_SIZE+1`, as the PSA format may change. We want to always use the PSA format as input to this function (so it makes sense to use its size macros when checking input), and always use `ATCA_PUB_KEY_SIZE` as output to the SE (when checking that after we've transformed the input to the SE's format, it's the correct size for the SE.)

Reply:
But you're speaking about the size of a buffer for a signature using a given key (and for our case it's `PSA_ASYMMETRIC_SIGN_OUTPUT_SIZE(PSA_KEY_TYPE_ECC_PUBLIC_KEY(PSA_ECC_CURVE_SECP256R1), 256, PSA_ALG_ECDSA(PSA_ALG_SHA_256))` == 64), not the size of a public key exported from PSA, which is 65. Maybe you've meant `PSA_KEY_EXPORT_ECC_PUBLIC_KEY_MAX_SIZE`?

Reply:
Yes, sorry. Wrong macro. Does PSA_KEY_EXPORT_ECC_PUBLIC_KEY_MAX_SIZE take an algorithm parameter? Is it overly large?

Reply:
It's short - added.